src/engine/strat_engine/setup.rs

```diff
@@ -26,6 +26,54 @@ use super::range_alloc::RangeAllocator;
 use super::serde_structs::PoolSave;
 
 
+/// Determine if devnode is a stratis device, if it is we will add  pool uuid and device
+/// information to pool_map and return pool uuid.
+pub fn is_stratis_device(devnode: &PathBuf) -> EngineResult<Option<PoolUuid>> {
+
+    let f = OpenOptions::new().read(true).open(&devnode);
```

Comment:
At this point this code really only has two branches, f err and f not, so might as well just turn the whole method into one big match expression. That way, you get to avoid the ```expect```s.